proto/akash/market/lease.proto

```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+package akash.market;
+
+import "gogoproto/gogo.proto";
+import "cosmos/cosmos.proto";
+
+option go_package = "github.com/ovrclk/akash/x/market/types";
+
+// LeaseID stores bid details of lease
+message LeaseID {
+    option (gogoproto.equal) = true;
+    option (gogoproto.goproto_stringer) = false;
+
+    bytes owner = 1 [
+        (gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress",
+        (gogoproto.jsontag)  = "owner",
+        (gogoproto.moretags) = "yaml:\"owner\""
+    ];
+    uint64 dseq = 2 [
+        (gogoproto.customname) = "DSeq",
+        (gogoproto.jsontag)    = "dseq",
+        (gogoproto.moretags)   = "yaml:\"dseq\""
+    ];
+    uint32 gseq = 3 [
+        (gogoproto.customname) = "GSeq",
+        (gogoproto.jsontag)    = "gseq",
+        (gogoproto.moretags)   = "yaml:\"gseq\""
+    ];
+    uint32 oseq = 4 [
+        (gogoproto.customname) = "OSeq",
+        (gogoproto.jsontag)    = "oseq",
+        (gogoproto.moretags)   = "yaml:\"oseq\""
+    ];
+    bytes provider = 5 [
+        (gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress",
+        (gogoproto.jsontag)  = "provider",
+        (gogoproto.moretags) = "yaml:\"provider\""
+    ];
+}
+
+// Lease stores LeaseID, state of lease and price
+message Lease {
+    option (gogoproto.equal) = false;
+
+    LeaseID lease_id = 1 [
+        (gogoproto.customname) = "LeaseID",
+        (gogoproto.nullable)   = false,
+        (gogoproto.jsontag)    = "id",
+        (gogoproto.moretags)   = "yaml:\"id\""
+    ];
+    uint32 state = 2 [
```

Comment:
can we use an enum for these?

Reply:
Yes, makes sense. Let's use enum for these